tests/test_opentracing.py

```diff
@@ -0,0 +1,517 @@
+# Here we monkey-patch tornado/testing.py with the fix from
+#   https://github.com/tornadoweb/tornado/pull/1574
+def patched_bind_unused_port(reuse_port=False):
+    sock = netutil.bind_sockets(None, '127.0.0.1', family=socket.AF_INET,
+                                reuse_port=reuse_port)[0]
+    port = sock.getsockname()[1]
+    return sock, port
+
+
+tornado.testing.bind_unused_port = patched_bind_unused_port
+
+
+@pytest.fixture
+def span_recorder():
+    class InMemoryRecorder(SpanRecorder):
+        def __init__(self):
+            self.spans = []
+            self.mux = threading.Lock()
+
+        def record_span(self, span):
+            with self.mux:
+                self.spans.append(span)
+
+        def get_spans(self):
+            with self.mux:
+                return self.spans[:]
+
+    return InMemoryRecorder()
+
+
+# noinspection PyShadowingNames
+@pytest.fixture
+def tracer(span_recorder):
+    return BasicTracer(recorder=span_recorder)
+
+
+class MockRequestContext(RequestContext):
+    """
+    This class extends the default `opentracing_instrumentation.RequestContext`
+    with a `parent_tracing` attribute, making it suitable for both TChannel
+    and `opentracing_instrumentation`.
+    """
+    __slots__ = ('parent_tracing',)
+
+    def __init__(self, span, parent_tracing=None):
+        super(MockRequestContext, self).__init__(span=span)
+        self.parent_tracing = parent_tracing
+
+
+class OpenTracingRequestContextProvider(RequestContextProvider):
+    """
+    Extends TChannel's RequestContextProvider by substituting the internal
+    `RequestContext` with `opentracing_instrumentation.RequestContext`.
+    """
+
+    def get_current_context(self):
+        """Returns a Context that has parent_tracing attribute"""
+        context = RequestContextManager.current_context()
+        if not context:
+            return None
+        if not hasattr(context, 'parent_tracing'):
+            context = MockRequestContext(
+                span=context.span,
+                parent_tracing=SpanWrapper(span=context.span)
+            )
+        return context
+
+    def request_context(self, parent_tracing):
+        """
+        Extracts Span from parent_tracing and captures it in StackContext
+        :param parent_tracing:
+        """
+        span = getattr(parent_tracing, 'span', None)
+        context = MockRequestContext(span=span, parent_tracing=parent_tracing)
+        # TODO should this be using a thread-safe version of StackContext?
+        return StackContext(lambda: RequestContextManager(context=context))
+
+
+@pytest.yield_fixture
+def http_patchers():
+    install_patches.__original_func()  # install_patches func is a @singleton
+    try:
+        yield 'http_patchers'
+    finally:
+        reset_patchers()
+
+
+def register(tchannel, http_client=None, base_url=None):
+    @tchannel.raw.register('endpoint1')
+    @tornado.gen.coroutine
+    def handler1(request):
+        print 'handler1 hit'
```

Comment:
they are quite helpful if the test is mis-behaving. When running `make test` these do not show up as long as the tests are successful, so is there any harm in keeping them?